### PR TITLE
Gibdo

### PR DESCRIPTION
**Pull request by Macrepeh**
Decompile Gibdo

### CODE REVIEW

File: src/enemy/gibdo.c [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+}
+END_NONMATCH
+
+void sub_080379BC(Entity* this) {
+    u32 r;
+    this->actionDelay = 0x18;
+    this->field_0xf = 0x8;
+    r = Random();
+    this->animationState = ((this->animationState + (r & 2)) - 1) & 3;
+    ;
```
Comment (notyourav) [inline src/enemy/gibdo.c:321]:
```suggestion
```

File: src/enemy/gibdo.c [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+    }
+    return 0;
+}
+
+void sub_08037A14(Entity* this) {
+    u32 x;
+    this->action = 6;
+    this->actionDelay = 0x18;
+    this->spritePriority.b0 = (this->spritePriority.b0 & (this->actionDelay - 0x20));
+    x = 3;
```
Comment (notyourav) [inline src/enemy/gibdo.c:342]: Is this temp needed?

File: src/enemy/gibdo.c [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+    this->field_0x3e = (this->animationState << 3) ^ 0x10;
+    this->field_0x42 = 8;
+    this->field_0x46 = 0x180;
+    this->field_0x76.HALF.LO = 0x3c;
+    InitAnimationForceUpdate(this, this->animationState + 0x10);
+}
+END_NONMATCH
+
+void sub_08037ACC(Entity* this) {
+    gPlayerState.flags.all &= 0xFFFFFEFF;
```
Comment (notyourav) [inline src/enemy/gibdo.c:370]:
```suggestion
    gPlayerState.flags.all &= ~0x100;
```

File: src/enemy/gibdo.c [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+END_NONMATCH
+
+void nullsub_162() {
+}
+
+void sub_08037558(Entity* this) {
+    u32 r;
+    sub_0804A720(this);
+    r = Random();
+    this->animationState = r & 0x3;
```
Comment (notyourav) [inline src/enemy/gibdo.c:87]: is this temp needed

File: src/enemy/gibdo.c [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+        if (sub_08037914(this) == 0) {
+            this->field_0x74.HWORD--;
+            if (this->field_0x74.HWORD == 0) {
+                sub_08037794(this);
+            } else {
+                UpdateAnimationSingleFrame(this);
+                UpdateAnimationSingleFrame(this);
+                if (ProcessMovement(this) == 0) {
+                    this->field_0xf--;
+                    if (this->field_0xf == 0) {
```
Comment (notyourav) [inline src/enemy/gibdo.c:142]: these and similar ones can usually be written as simply `if (--this->field_0xf)`

File: src/enemy/gibdo.c [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+    }
+}
+
+void sub_08037690(Entity* this) {
+    UpdateAnimationSingleFrame(this);
+    if ((this->frames.all & 0x80) != 0) {
+        this->field_0x76.HALF.HI = 0x14;
+        sub_08037794(this);
+    } else {
+        if ((this->frames.all & 1) != 0) {
```
Comment (notyourav) [inline src/enemy/gibdo.c:162]:
```suggestion
        if (this->frames.all & 1) {
```
can be simplified

File: src/enemy/gibdo.c [hunk 7]
```diff
@@ -0,0 +1,434 @@
...
+void sub_080377B0(Entity* this) {
+    u32 r1;
+    u32 r2;
+    this->action = 2;
+    this->field_0xf = 8;
+    r1 = Random();
+    this->field_0x74.HWORD = (r1 & 0x38) + 0x78;
+    this->nonPlanarMovement = 0x40;
+    r2 = Random();
+    if ((sub_08049FA0(this) == 0) && ((r2 & 3) != 0)) {
```
Comment (notyourav) [inline src/enemy/gibdo.c:227]:
```suggestion
    if (!sub_08049FA0(this) && (r2 & 3)) {
```

File: src/enemy/gibdo.c [hunk 8]
```diff
@@ -0,0 +1,434 @@
...
+        this->direction = (sub_08049EE4(this) + 4) & 0x18;
+    } else {
+        this->direction = r2 & 0x18;
+    }
+    this->animationState = this->direction / 8;
```
Comment (notyourav) [inline src/enemy/gibdo.c:232]: Many of these can be turned into proper macros.
see https://github.com/zeldaret/tmc/blob/master/include/entity.h#L202

**Review by notyourav — CHANGES_REQUESTED** [review]

File: src/enemy/gibdo.c (hunk 2 above)
Comment (Macrepeh) [reply src/enemy/gibdo.c:342]: It isn't I just checked sorry about that

File: src/enemy/gibdo.c (hunk 1 above)
Comment (Macrepeh) [reply src/enemy/gibdo.c]: Oopsie

**Review by notyourav — APPROVED** [review]

**Pull request merged**